include/storage/shared_monitor.hpp

```diff
@@ -42,7 +42,16 @@ template <typename Data> struct SharedMonitor
 
     SharedMonitor(const Data &initial_data)
     {
-        shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
+        try
+        {
+            shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
```

Comment:
@mjjbell I saw you had concerns about this PR. Is it okay now? I could probably rebase and merge it

Reply:
I can't remember exactly. I think it looked unfinished.